controller/scheduler-v2/scheduler.go

```diff
@@ -936,49 +995,73 @@ func (s *Scheduler) Jobs() map[string]*Job {
 	return jobs
 }
 
-func (s *Scheduler) expandOmni(ef *ct.ExpandedFormation) {
-	release := ef.Release
-
-	for typ, pt := range release.Processes {
-		if pt.Omni && ef.Processes != nil && ef.Processes[typ] > 0 {
-			ef.Processes[typ] *= len(s.hostStreams)
-		}
+func (s *Scheduler) scheduleJobRequest(req *JobRequest) error {
+	log := fnLogger()
+	if req.startedAt.Before(time.Now().Add(-s.backoffPeriod)) {
+		log.Info("resetting job restarts", "backoffPeriod", s.backoffPeriod, "job.startedAt", req.startedAt)
+		req.restarts = 0
 	}
-}
-
-func (s *Scheduler) scheduleJobRequest(req *JobRequest) {
 	backoff := s.getBackoffDuration(req.restarts)
+	if backoff == s.maxBackoffPeriod {
```

Comment:
Don't use exponential backoff past the max period (`10 * s.backoffPeriod`). After that, the attempt will fail and we'll have to rely on `Rectify`.